src/main/java/org/veupathdb/service/eda/ss/model/variable/binary/BinaryFilesManager.java

```diff
@@ -114,6 +122,34 @@ public Path getDoneFile(Path directory, Operation op) {
     return filepath;
   }
 
+  /**
+   * For a given entity, parse the metadata JSON file to determine the number of bytes needed to store identifiers
+   * for all the entity's ancestors.
+   * @return a list of ancestors in order of closest to the farthest ascendant.
+   */
+  public List<Integer> getBytesReservedForAncestry(Study study, Entity entity) {
+    JSONObject metajson = readMetaJsonFile(getMetaJsonFile(study, entity, Operation.READ));
+    getMetaJsonFile(study, entity, Operation.READ);
```

Comment:
why are we calling getMetaJsonFile() again?

should getBytesReservedForAncestry be memoized/cached?